src/libcore/panicking.rs

```diff
@@ -48,16 +48,24 @@ pub fn panic(expr_file_line_col: &(&'static str, &'static str, u32, u32)) -> ! {
     // truncation and padding (even though none is used here). Using
     // Arguments::new_v1 may allow the compiler to omit Formatter::pad from the
     // output binary, saving up to a few kilobytes.
-    let (expr, file, line, col) = *expr_file_line_col;
-    panic_fmt(fmt::Arguments::new_v1(&[expr], &[]), &(file, line, col))
+    if cfg!(feature = "panic_immediate_abort") {
+        unsafe { super::intrinsics::abort() }
+    } else {
```

Comment:
I think it's probably fine to leave out the `else` clause here, and just rely on the fact that the `if` above won't actually continue if it executes

Reply:
Why leave out `else`? How would it work without `panic_immediate_abort` then?

I expect either "then" branch or "else" branch of `if cfg!()` to be always removed by optimizer.

Or maybe I fail to understand the comment?..

Reply:
Oh sorry, I mean instead of this:

```rust
if foo {
    a;
} else {
    b;
}
```

to do:

```rust
if foo {
    a;
}

b;
```